### PR TITLE
INTMDB-400: Feature add: Enable support for Snapshot Distribution

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

Enable support for Snapshot Distribution feature

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/data_source_mongodbatlas_cloud_backup_schedule.go [hunk 1]
```diff
@@ -25,6 +25,37 @@ func dataSourceMongoDBAtlasCloudBackupSchedule() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"copy_settings": {
```
Comment (Zuhairahmed) [inline mongodbatlas/data_source_mongodbatlas_cloud_backup_schedule.go:28]: @martinstibbe can you confirm that we don't need to make any updates to [go-client](https://github.com/mongodb/go-client-mongodb-atlas)?

File: website/docs/r/cloud_backup_schedule.html.markdown [hunk 2]
```diff
@@ -181,6 +181,13 @@ resource "mongodbatlas_cloud_backup_schedule" "test" {
 * `retention_unit` - (Required) Scope of the backup policy item: days, weeks, or months.
 * `retention_value` - (Required) Value to associate with `retention_unit`.
 
+### Snapshot Distribution
+*
+* `cloud_provider` - (Required) Human-readable label that identifies the cloud provider that stores the snapshot copy. i.e. "AWS" "AZURE" "GCP"
```
Comment (Zuhairahmed) [inline website/docs/r/cloud_backup_schedule.html.markdown:234]: Can you add an example demonstrating how to use this feature on this page?

File: website/docs/d/cloud_backup_schedule.html.markdown [hunk 3]
```diff
@@ -101,4 +101,12 @@ In addition to all arguments above, the following attributes are exported:
 * `retention_unit` - Scope of the backup policy item: days, weeks, or months.
 * `retention_value` - Value to associate with `retention_unit`.
 
+### Snapshot Distribution
```
Comment (Zuhairahmed) [inline website/docs/d/cloud_backup_schedule.html.markdown:104]: Can you add an example demonstrating how to use this feature on this page?

File: mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go [hunk 4]
```diff
@@ -60,6 +60,42 @@ func resourceMongoDBAtlasCloudBackupSchedule() *schema.Resource {
 				Optional: true,
 				Computed: true,
 			},
+			"copy_settings": {
```
Comment (Zuhairahmed) [inline mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go:63]: @martinstibbe can you double check that we are also supporting "deleteCopiedBackups" parameters as part of this update? https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Cloud-Backup-Schedule/operation/updateCloudBackupBackupPolicyForOneCluster

**Review by Zuhairahmed — CHANGES_REQUESTED** [review]
Few comments below, this is a good start!

File: mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go [hunk 5]
```diff
@@ -414,6 +454,12 @@ func cloudBackupScheduleCreateOrUpdate(ctx context.Context, conn *matlas.Client,
 	var policiesItem []matlas.PolicyItem
 	export := matlas.Export{}
 
+	if v, ok := d.GetOk("copy_settings"); ok && len(v.([]interface{})) > 0 {
+		req.CopySettings = expandCopySettings(v.([]interface{}))
+	} else {
+		req.CopySettings = []matlas.CopySetting{}
+	}
+
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go]:
```suggestion
	req.CopySettings = []matlas.CopySetting{}
	if v, ok := d.GetOk("copy_settings"); ok && len(v.([]interface{})) > 0 {
		req.CopySettings = expandCopySettings(v.([]interface{}))
	} 
```

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: mongodbatlas/data_source_mongodbatlas_cloud_backup_schedule.go (hunk 1 above)
Comment (martinstibbe) [reply mongodbatlas/data_source_mongodbatlas_cloud_backup_schedule.go:28]: copy_settings is supported in go-client
`type CloudProviderSnapshotBackupPolicy struct {
	.....
	CopySettings                      []CopySetting        `json:"copySettings"`                                // List that contains a document for each copy setting item in the desired backup policy.
	DeleteCopiedBackups               []DeleteCopiedBackup `json:"deleteCopiedBackups,omitempty"`               // List that contains a document for each deleted copy setting whose backup copies you want to delete.
}
`

File: website/docs/r/cloud_backup_schedule.html.markdown (hunk 2 above)
Comment (martinstibbe) [reply website/docs/r/cloud_backup_schedule.html.markdown:234]: Added

File: mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go (hunk 5 above)
Comment (martinstibbe) [reply mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go]: removed if else structure

File: website/docs/d/cloud_backup_schedule.html.markdown (hunk 3 above)
Comment (themantissa) [reply website/docs/d/cloud_backup_schedule.html.markdown:104]: FYI - Normally the example will be in the resource, not data source, unless it's data source specific.

File: mongodbatlas/data_source_mongodbatlas_cloud_backup_schedule.go (hunk 1 above)
Comment (Zuhairahmed) [reply mongodbatlas/data_source_mongodbatlas_cloud_backup_schedule.go:28]: sounds good!

File: mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go (hunk 4 above)
Comment (Zuhairahmed) [reply mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go:63]: @martinstibbe are you planning to add this additional parameter in?

Comment (martinstibbe) [reply mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go:63]: This parameter will be skipped as it does not fit usage pattern in TF (more of a backup pruning schedule function)

Comment (Zuhairahmed) [reply mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go:63]: ok thanks. can you call this out in docs explicitly then (just to get ahead as i'm sure customers are going to ask in future) that "deleteCopiedBackups" is not supported in TF and if they need can leverage in the Atlas Admin API or AtlasCLI instead.

Comment (martinstibbe) [reply mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go:63]: Added Note to docs

Comment (Zuhairahmed) [reply mongodbatlas/resource_mongodbatlas_cloud_backup_schedule.go:63]: Looks good!

**Review by Zuhairahmed — APPROVED** [review]
LGTM! @martinstibbe thanks for all the efforts on this one, customers are going to love this feature add!

**Pull request merged**